packages/web/src/scss/components/Header/_HeaderDialog.scss

```diff
@@ -0,0 +1,116 @@
+// 1. Use a `::before` pseudo-element instead of `::backdrop` to make fading possible.
+// 2. HeaderDialog experimentally supports RTL.
+// 3. Use an `::after` pseudo-element to create a box shadow that is not rounded in corners.
+// 4. While other elements get spacing inside action slots, the nav needs to take up the whole width of the pane.
+
+@use '../../tools/typography';
+@use 'theme';
+
+.HeaderDialog {
+    @include typography.generate(theme.$header-dialog-typography);
+
+    position: fixed;
+    inset: 0;
+    z-index: 1;
```

Comment:
It remember to me, that we talked about the sharing z-indeces across the components – candidate on the Great Topics 🥌.

Meanwhile we use `z-index` instead `isolation: isolate` (honestly, I'm not sure if in this case it would make any sense), what about to define these values in the `_theme.scs` too? And [demonstrate relationship](https://github.com/sfrieson/managing-css-z-index-in-large-projects/blob/main/language-styles/z-index-constants.scss) between z-indeces between blocks?

I rather don't talk about [mixin for stacking context](https://www.smashingmagazine.com/2021/02/css-z-index-large-projects/#comment-1612885778105196393) – it looks little bit like _a cannon on a sparrow_. 🙂

Reply:
It's not as easy as it might appear, I'm afraid… There are at least several types of stacking:

1. locally scoped stacking within a component,
2. global layers like header, app-loader, off-canvas drawer, maybe dialogs,
3. context-relative stacking like tooltips or dropdowns.

To share, or not to share?

1. There is no reason to share anything because everything is local. This is where `isolation: isolate` can be useful.
2. There are not many suitable components, but we can certainly identify some of them, e.g. the Header.
3. This is where `z-index: 1` probably just works.

Reply:
OK, let's talk about the concept of z-indices later.